### DISCUSSION TITLE
Extended built-ins

### DISCUSSION THREAD

**Discussion by LeaVerou** [Devographics/surveys#99] [State of HTML Features]
Extend built-in elements with custom behavior.

```html
<button is="primary-button">Continue</button>
<script>
	customElements.define("primary-button", PrimaryButton, {extends: "button"});
</script>
```

### MDN URL:

https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/is

### Specification URL:

https://html.spec.whatwg.org/multipage/custom-elements.html#attr-is

### Any other links?

- https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/define
- [WebKit Standards Position](https://github.com/WebKit/standards-positions/issues/97)

### How stable is this feature?

Controversial (Some implementations that have existed for years, other browsers have declined to implement)

### Type of feature

HTML attribute, JS API

### What focus area does this feature fall into?

Consuming Web Components, Making Web Components

### Details

We could ask about the `is` attribute and `customElementRegistry.define("tagname", Class, {extends})` separately if we want to distinguish consumption from usage.
For now, I’ve included them in the same item.

**Comment by colepeters** [comment]
May be worth adding this to 'other links', as WebKit is declining to implement this feature: https://github.com/WebKit/standards-positions/issues/97

**Reply by LeaVerou** [reply]
Done! (though note I had already listed it as "Controversial" because of this)
